tools/challenge-parser/parser/plugins/add-video-question.js

```diff
@@ -9,17 +9,37 @@ const { splitOnThematicBreak } = require('./utils/split-on-thematic-break');
 function plugin() {
   return transformer;
   function transformer(tree, file) {
-    const questionNodes = getAllBetween(tree, '--question--');
-    if (questionNodes.length > 0) {
-      const questionTree = root(questionNodes);
+    const allQuestionNodes = getAllBetween(tree, '--questions--');
 
-      const textNodes = getAllBetween(questionTree, '--text--');
-      const answersNodes = getAllBetween(questionTree, '--answers--');
-      const solutionNodes = getAllBetween(questionTree, '--video-solution--');
+    if (allQuestionNodes.length > 0) {
+      const questions = [];
+      const questionTrees = [];
+      let questionTreeNumber = -1;
 
-      const question = getQuestion(textNodes, answersNodes, solutionNodes);
+      allQuestionNodes.forEach(questionNode => {
+        if (
+          questionNode.children &&
+          questionNode.children[0] &&
+          questionNode.children[0].value === '--text--'
+        ) {
+          questionTreeNumber++;
+          questionTrees[questionTreeNumber] = [];
+        }
 
-      file.data.question = question;
+        questionTrees[questionTreeNumber].push(questionNode);
+      });
+
```

Comment:
```suggestion

      allQuestionNodes.forEach(questionNode => {
        const isStartOfQuestion =
          questionNode.children?.[0]?.value === '--text--';
        if (isStartOfQuestion) {
          questionTrees.push([questionNode]);
        } else {
          questionTrees[questionTrees.length - 1].push(questionNode);
        }
      });

```
I think this is a little cleaner and self-descriptive.

Reply:
I think this is the right place, but reviewing through VSCode is horrendous and reviewing through the browser is impossible :(

Anyways, it's supposed to replace the code that populates the `questionTrees`

Reply:
✅